torch/_inductor/autotune_process.py

```diff
@@ -42,6 +43,27 @@ class Pong:
     pass
 
 
+@contextlib.contextmanager
+def set_cuda_visible_device(device: Optional[int]):
+    """
+    Context manager to set the CUDA_VISIBLE_DEVICES environment variable to the
+    specified single device. If device is None, don't manipulate the environment.
+    """
+    if device is None:
+        yield
```

Comment:
To double check my understanding: if the user sets `CUDA_VISIBLE_DEVICES=7` on the parent python process running `torch.compile` and multi-device autotuning is *disabled*, we follow this code path, but the env var value should propagate to the child process. Is this correct?

Reply:
Thanks for the detailed explanation! This looks correct.